ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaViewModel.kt

```diff
@@ -75,22 +85,49 @@ class AddMediaViewModel(application: Application) : AndroidViewModel(application
     }
 
     // We create a URI where the camera will store the video
-    fun createVideoUri(): Uri? {
+    fun createVideoUri(source: Source): Uri? {
         val videoCollection = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
             MediaStore.Video.Media.getContentUri(MediaStore.VOLUME_EXTERNAL_PRIMARY)
         } else {
             MediaStore.Video.Media.EXTERNAL_CONTENT_URI
         }
 
         val newVideo = ContentValues().apply {
-            put(MediaStore.Video.Media.DISPLAY_NAME, generateFilename(Source.CAMERA, "mp4"))
+            put(MediaStore.Video.Media.DISPLAY_NAME, generateFilename(source, "mp4"))
         }
 
         val uri = context.contentResolver.insert(videoCollection, newVideo)
         temporaryMediaUri = uri
 
         return uri
     }
+
+    fun saveRandomImageFromInternet() {
+        viewModelScope.launch {
+            val imageUri = createPhotoUri(Source.INTERNET)
+            val request = Request.Builder().url(RANDOM_IMAGE_URL).build()
+
+            withContext(Dispatchers.IO) {
+                val response = httpClient.newCall(request).execute()
+
+                response.body?.let { responseBody ->
+                    imageUri?.let { destinationUri ->
+                        context.contentResolver.openOutputStream(destinationUri, "w")?.let {
+                            responseBody.byteStream().copyTo(it)
+
+                            // Once we save the image, we need to set the_currentMediaUri. We can't
+                            // use `loadCameraMedia` as updating the MutableLiveData in a background
+                            // thread requires to use `postValue`
+                            _currentMediaUri.postValue(temporaryMediaUri)
```

Comment:
`temporaryMediaUri` is read from and written to from multiple threads without explicit synchronization. As far as I can tell this isn't a problem right now because there's no concurrent access and coroutine magic will make sure changes are observable by other threads. Still, it's probably a good idea to only access this property from the main thread.